extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -92,27 +91,55 @@ export class SubscriptionService {
   registerService(serviceId, subscriptionPlatform) {
     this.subscriptionPlatforms_.push(subscriptionPlatform);
 
-    subscriptionPlatform.getEntitlements()
-        .then(() => this.processEntitlement_());
+    this.fetchEntitlements_(subscriptionPlatform);
   }
 
-  /** @private */
-  processEntitlement_() {
-    // TODO(@prateekbh): process and unblock marup here.
+  /**
+   * @param {boolean} grantState
+   * @private
+   */
+  processEntitlements_(grantState) {
+    if (grantState === false) {
+      // TODO(@prateekbh): Show UI that no eligible entitlement found
+      return;
+    }
+
+    this.renderer_.toggleLoading(false);
```

Comment:
Need to toggle off the loading indicator even with the `grantState === false`, no?

Reply:
yeah I wasn't really sure what to do there, but yeah makes sense